### PR TITLE
[PTT-94] Provision Frontend

### PR DESCRIPTION
**Pull request by GiovanniMancini**
##Description
This PR proposes to add a module `frontend` based on a `blob storage` and a `cdn endpoint` to deploy a frontend based on static web files.

## Modules Added
- A module for a blob storage to deploy and serve static web files
- A module for a cdn endpoint
- A module `frontend` which can add a new blob storage configured to host static web content and the related cdn endpoint

## Resources added
- A new resource group
- A blob storage
- A cdn profile
- A cdn endpoint

## Status
Provisioned

### CODE REVIEW

File: modules/blob_storage/main.tf [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    }
+  }
+
+  tags = var.tags
+}
+
+# Enable advanced threat protection
+resource "azurerm_advanced_threat_protection" "advanced_threat_protection" {
+  target_resource_id = azurerm_storage_account.storage_account.id
+  enabled            = true
```
Comment (uolter) [inline modules/blob_storage/main.tf:63]: Why do you enable by default the threat protection? Keep in mind it has "considerable" costs this setting.
Whether the storage contains static / public content might not need this additional property

File: modules/blob_storage/main.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+
+
+# this is a tempory implementation till an official one will be released:
+# https://github.com/terraform-providers/terraform-provider-azurerm/issues/8268
+
+resource "azurerm_template_deployment" "versioning" {
```
Comment (uolter) [inline modules/blob_storage/main.tf:71]: As the advanced thread protection version has addition costs. I would not enable it by default.

File: modules/cdn_endpoint/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+output "id" {
+  value = azurerm_cdn_endpoint.cdn_endpoint.id
+}
+
+output "hostname" {
+  value = "${var.name}.azureedge.net"
```
Comment (uolter) [inline modules/cdn_endpoint/outputs.tf:6]:
```suggestion
  value = join(".", [var.name, "azureedge.net" ])
```

File: outputs.tf [hunk 4]
```diff
@@ -35,4 +35,20 @@ output "api_gateway_public_id" {
...
+
+output "frontend_cdn_endpoint_name"  {
+  value = module.fe.cdn_endpoint_name
+}
+
+output "frontend_cdn_profile_name"  {
+  value = azurerm_cdn_profile.common.name
+}
+
+output "rg_frontend" {
```
Comment (uolter) [inline outputs.tf:52]: Is this output resource group usefutl?

**Review by uolter — CHANGES_REQUESTED** [review]

File: modules/blob_storage/main.tf (hunk 1 above)
Comment (GiovanniMancini) [reply modules/blob_storage/main.tf:63]: Ok I'll switch to false

File: outputs.tf (hunk 4 above)
Comment (GiovanniMancini) [reply outputs.tf:52]: I need it in `gitops` repo to configure deploy pipeline

File: modules/cdn_endpoint/outputs.tf (hunk 3 above)
Comment (GiovanniMancini) [reply modules/cdn_endpoint/outputs.tf]: Fixed

**Review by uolter — APPROVED** [review]

**Pull request merged**